actors/builtin/storage_power/storage_power_test.go

```diff
@@ -1 +1,128 @@
 package storage_power_test
+
+import (
+	"context"
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/ipfs/go-cid"
+	"github.com/stretchr/testify/require"
+	"testing"
+
+	addr "github.com/filecoin-project/go-address"
+	builtin "github.com/filecoin-project/specs-actors/actors/builtin"
+	storage_power "github.com/filecoin-project/specs-actors/actors/builtin/storage_power"
+	adt "github.com/filecoin-project/specs-actors/actors/util/adt"
+	mock "github.com/filecoin-project/specs-actors/support/mock"
+	assert "github.com/stretchr/testify/assert"
+)
+
+func TestConstruction(t *testing.T) {
+	actor := spActorHarness{storage_power.StoragePowerActor{}, t}
+	powerActor := newIDAddr(t, 100)
+	owner1 := newIDAddr(t, 101)
+	worker1 := newIDAddr(t, 102)
+	miner1 := newIDAddr(t,103)
+
+	builder := mock.NewBuilder(context.Background(), powerActor).WithCaller(builtin.SystemActorAddr, builtin.SystemActorCodeID)
+
+	t.Run("simple construction", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+	})
+
+	t.Run("create miner", func(t *testing.T) {
+		createMinerParams := &storage_power.CreateMinerParams{
+			Worker: worker1,
+			SectorSize: abi.SectorSize(int64(32)),
+			Peer: "miner1",
+		}
+
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		// owner1 send CreateMiner to StoragePowerActor
+		rt.SetCaller(owner1, builtin.AccountActorCodeID)
+		rt.SetReceived(big.Zero())
```

Comment:
Set a non-zero amount here and verify it gets accounted as pledge collateral  for the miner